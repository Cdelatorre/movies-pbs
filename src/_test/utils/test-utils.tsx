import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { AppStore, RootState, setupStore } from "../../store";
import { BrowserRouter, Router } from "react-router-dom";
import { emptyStoreData } from "./mocks/mock";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = emptyStoreData,

    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  route?: string
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    if (route) window.history.pushState({}, "Test routing", route);

    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>;
      </BrowserRouter>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderWithRouter(ui: React.ReactElement, history?: any) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    if (history) {
      return (
        <Router location="/" navigator={history}>
          {children}
        </Router>
      );
    } else {
      return <BrowserRouter>{children}</BrowserRouter>;
    }
  }
  return { ...render(ui, { wrapper: Wrapper }) };
}
