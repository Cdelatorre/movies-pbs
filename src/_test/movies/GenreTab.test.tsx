import { fireEvent, screen } from "@testing-library/react";
import GenreTab from "../../components/movies/genre-tab";
import { renderWithRouter } from "../utils/test-utils";

describe("Renders GenreTab component correctly", () => {
  test("Renders default GenreTab component correctly", () => {
    renderWithRouter(<GenreTab id="Genre">Genre</GenreTab>);

    const genreTab = screen.getByTestId("genre-tab-Genre");

    expect(genreTab).toBeInTheDocument();
    expect(screen.getByText("Genre")).toBeInTheDocument();

    expect(genreTab).toMatchSnapshot();
  });

  test("Renders delete action button and calls onDelete function when clicked", () => {
    const onDelete = jest.fn();
    renderWithRouter(
      <GenreTab id="Genre" onDelete={onDelete}>
        Genre
      </GenreTab>
    );

    const deleteBtn = screen.getByTestId("genre-delete-Genre");

    expect(deleteBtn).toBeInTheDocument();
    fireEvent.click(deleteBtn);

    expect(onDelete).toHaveBeenCalledTimes(1);
  });
});
