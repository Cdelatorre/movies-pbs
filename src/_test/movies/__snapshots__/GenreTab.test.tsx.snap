// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders GenreTab component correctly Renders default GenreTab component correctly 1`] = `
<span
  data-testid="genre-tab-Genre"
>
  Genre
</span>
`;
